src/lib.rs

```diff
@@ -432,14 +442,16 @@ fn identify_f<C: Container>(
         )?;
 
         // Make sure the spheres are bounded by the containing geometry and do not overlap any spheres in V
-        if container.contains(&s_4_positive) && !set_v.iter().any(|v| v.overlaps(&s_4_positive)) {
-            f.push(s_4_positive);
+        if container.contains(&s_4_positive) && !set_v.par_iter().any(|v| v.overlaps(&s_4_positive))
```

Comment:
Dropping this `par_iter` and the one directly below gives me 18 seconds on the first benchmark and 2 mins 47 secs on the second. So not that much improvement yet...